lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php

```diff
@@ -218,6 +218,30 @@ public function setTemplateOptions($templateOptions)
     }
 
     /**
+     * Creates a Zend_Mime_Part attachment and attaches it to the current message scope.
+     * Attachment is automatically added to the mail object after creation.
+     *
+     * See also: http://framework.zend.com/manual/1.12/de/zend.mail.attachments.html
+     *
+     * @param  string $body
+     * @param  string $mimeType
+     * @param  string $disposition
+     * @param  string $encoding
+     * @param  string $filename A filename for the attachment
+     * @return $this
+     */
+    public function addAttachment(
+        $body,
+        $mimeType = \Zend_Mime::TYPE_OCTETSTREAM,
+        $disposition = \Zend_Mime::DISPOSITION_ATTACHMENT,
+        $encoding = \Zend_Mime::ENCODING_BASE64,
+        $filename = null
+    ) {
+        $this->message->createAttachment($body, $mimeType, $disposition, $encoding, $filename);
```

Comment:
> First, I do not understand why this should be a backward incompatible change due to the only extensible kind of change and no change on the existing code (new method, no override).

When you add or remove a new method on an interface, you're breaking _every_ other 3rd party implementation of that interface. The high level API is a hard thing to change without pissing people off.

@vkublytskyi is not refering to _internal_ classes, but _other people's_ implementation of that interface.

And I agree with him that you're leaking implementation details (`\Zend_Mime`). If I want it to be as abstract as possible (as it should be in an interface/API), I usually try to test it with 3 different implementations (e.g. Zend, Symfony etc.).